### PR TITLE
Feature/run web

### PR DESCRIPTION
**Pull request by dpocheronOnePoint**

### CODE REVIEW

File: lib/injection/data_source_injection.dart [hunk 1]
```diff
@@ -1,12 +1,20 @@
+// import 'package:api/data_source/tree/local_tree_data_source.dart';
```
Comment (Hamitix) [inline lib/injection/data_source_injection.dart]: Comment to delete

File: lib/injection/data_source_injection.dart [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
 import 'package:api/data_source/tree/local_tree_data_source.dart';
 import 'package:api/data_source/tree/remote_tree_data_source.dart';
 import 'package:data/data_source/tree/local_tree_data_source_impl.dart';
+import 'package:data/data_source/tree/local_tree_web_data_source_impl.dart';
 import 'package:data/data_source/tree/remote_tree_data_source_impl.dart';
 import 'package:api/dependency_injection.dart';
+import 'package:flutter/foundation.dart';
```
Comment (Hamitix) [inline lib/injection/data_source_injection.dart]: Import at the very top

File: lib/injection/managers_injection.dart [hunk 3]
```diff
@@ -3,11 +3,13 @@ import 'package:api/models/app/managers/remote_config.dart';
 import 'package:api/models/app/managers/crash.dart';
 
 import 'package:data/firebase/remote_config/firebase_remote_config.dart';
+import 'package:flutter/foundation.dart';
```
Comment (Hamitix) [inline lib/injection/managers_injection.dart]: Import at the very top

File: lib/screens/settings/settings_screen.dart [hunk 4]
```diff
@@ -65,16 +66,18 @@ class _SettingsScreenState extends State<SettingsScreen> {
...
+                Row(
+                  mainAxisAlignment: MainAxisAlignment.center,
+                  children: [
+                    Text(LocaleKeys.crash_manager_activated_label.localized()),
+                    Switch(
+                        activeColor: Theme.of(context).primaryColorDark,
+                        value: viewModel.isCrashManagerEnabled,
+                        onChanged: viewModel.toggleCrashManager),
+                  ],
+                ),
```
Comment (Hamitix) [inline lib/screens/settings/settings_screen.dart]: Logic to put in the viewModel

File: packages/data/lib/data_source/tree/local_tree_data_source_impl.dart [hunk 5]
```diff
@@ -12,15 +14,20 @@ class LocalTreeDataSourceImpl implements LocalTreeDataSource {
   @override
   List<Tree> getTreeList() {
     List<Tree> trees = [];
-    List<LocalTree> list = localStorage.getItems();
-    for (var item in list) {
-      trees.add(item.toDomain());
+    if (kIsWeb) {
```
Comment (Hamitix) [inline packages/data/lib/data_source/tree/local_tree_data_source_impl.dart]: Should be the opposite ? ( if (!kIsWeb) )

File: packages/data/lib/data_source/tree/local_tree_web_data_source_impl.dart [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  @override
+  List<Tree> getTreeList() {
+    return [];
+  }
+
+  @override
+  List<int> saveTreeList(List<Tree> list) {
+    return [];
+  }
+}
```
Comment (Hamitix) [inline packages/data/lib/data_source/tree/local_tree_web_data_source_impl.dart]: Replace ' { return ' with ' => '

File: packages/data/lib/local_storage/local_storage_web_impl.dart [hunk 7]
```diff
@@ -0,0 +1,40 @@
+import 'package:data/local_storage/local_storage.dart';
+
+class LocalStorageImpl implements LocalStorage {
+  LocalStorageImpl._create();
+
+  static Future<LocalStorageImpl> create() async {
+    return LocalStorageImpl._create();
+  }
+
```
Comment (Hamitix) [inline packages/data/lib/local_storage/local_storage_web_impl.dart]: Same for this file : Replace ' { return ' with ' => '

File: packages/domain/lib/use_case/tree/get_tree_list_use_case.dart [hunk 8]
```diff
@@ -1,4 +1,4 @@
-import 'package:api/data_source/tree/local_tree_data_source.dart';
+// import 'package:api/data_source/tree/local_tree_data_source.dart';
```
Comment (Hamitix) [inline packages/domain/lib/use_case/tree/get_tree_list_use_case.dart]: Comment to delete

File: packages/domain/lib/use_case/tree/get_tree_list_use_case.dart [hunk 9]
```diff
@@ -8,8 +8,8 @@ import 'package:api/use_case/tree/tree_list.dart';
 class GetTreeListUseCase implements GetTreeList {
   RemoteTreeDataSource remoteTreeDataSource =
       DependecyInjection.instance.get<RemoteTreeDataSource>();
-  LocalTreeDataSource localTreeDataSource =
-      DependecyInjection.instance.get<LocalTreeDataSource>();
+  // LocalTreeDataSource localTreeDataSource =
+  //     DependecyInjection.instance.get<LocalTreeDataSource>();
```
Comment (Hamitix) [inline packages/domain/lib/use_case/tree/get_tree_list_use_case.dart]: Comment to delete

File: packages/domain/lib/use_case/tree/get_tree_list_use_case.dart [hunk 10]
```diff
@@ -20,10 +20,11 @@ class GetTreeListUseCase implements GetTreeList {
       case FetchStrategy.remote:
         List<Tree> glTrees = await remoteTreeDataSource.getTreeListRemote(
             startRow: startRow, nbRows: nbRows);
-        localTreeDataSource.saveTreeList(glTrees);
+        // localTreeDataSource.saveTreeList(glTrees);
```
Comment (Hamitix) [inline packages/domain/lib/use_case/tree/get_tree_list_use_case.dart]: Comment to delete

File: assets/translations/en.json [hunk 11]
```diff
@@ -63,5 +63,7 @@
 
     "crash_manager" : {
         "activated_label" : "Crash manager is activated:"
-    }
+    },
+
+    "label_force_language_setting" : "langage is english"
```
Comment (Hamitix) [inline assets/translations/en.json]: This translation is not used right ?

Comment (dpocheronOnePoint) [reply assets/translations/en.json]: it's the translation for the firebase remote config key

File: lib/screens/settings/settings_view_model.dart [hunk 12]
```diff
@@ -3,6 +3,7 @@ import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
 
 import 'package:adaptive_theme/adaptive_theme.dart';
+import 'package:globo_fitness/extensions/string_replace_dot_remote_config.dart';
```
Comment (Hamitix) [inline lib/screens/settings/settings_view_model.dart]: Import not in the right place

File: lib/screens/settings/settings_view_model.dart [hunk 13]
```diff
@@ -88,10 +89,16 @@ abstract class SettingsViewModelBase with Store, ViewModel {
...
-      _remoteConfigManager.getValue<String>(
-          key: RemoteConfigKeys.key_label_force_language_setting.name) ??
-      "";
+  String getLabelChangeLanguage() {
+    String remoteKey = _remoteConfigManager.getValue<String>(
+            key: RemoteConfigKeys
+                .remote_config_key_label_force_language_setting.name) ??
+        "";
+
+    // It's needed to use LoacalKeys from String
```
Comment (Hamitix) [inline lib/screens/settings/settings_view_model.dart]: Typo in comment

File: lib/screens/settings/settings_view_model.dart [hunk 14]
```diff
@@ -104,6 +111,23 @@ abstract class SettingsViewModelBase with Store, ViewModel {
...
+              activeColor: Theme.of(context).primaryColorDark,
+              value: isCrashManagerEnabled,
+              onChanged: toggleCrashManager),
+        ],
+      );
+    } else {
+      return Row();
+    }
+  }
+
```
Comment (Hamitix) [inline lib/screens/settings/settings_view_model.dart]: 1) Changed the Function return type to 'Widget'. 
In the else case, return "const SizedBox.shrink()" (More efficient way to render a empty view).

File: lib/extensions/string_replace_dot_remote_config.dart [hunk 15]
```diff
@@ -0,0 +1,6 @@
+extension ReplaceDotInRemoteConfigString on String {
+  String replaceDotInRemoteConfig() {
+    String stringWithoutDot = replaceAll('remote_config_', 'remote_config.');
+    return stringWithoutDot;
+  }
+}
```
Comment (Hamitix) [inline lib/extensions/string_replace_dot_remote_config.dart:6]: Make a generic replaceUnderscoreToDot methods for possible future uses. ( use the 'this' to get the current string value)

File: lib/screens/settings/settings_view_model.dart [hunk 16]
```diff
@@ -88,10 +89,16 @@ abstract class SettingsViewModelBase with Store, ViewModel {
...
-      "";
+  String getLabelChangeLanguage() {
+    String remoteKey = _remoteConfigManager.getValue<String>(
+            key: RemoteConfigKeys
+                .remote_config_key_label_force_language_setting.name) ??
+        "";
+
+    // It's needed to use LoacalKeys from String
+    remoteKey = remoteKey.replaceDotInRemoteConfig().localized();
+    return remoteKey;
```
Comment (Hamitix) [inline lib/screens/settings/settings_view_model.dart]: Return directly  'remoteKey.replaceDotInRemoteConfig().localized();'

**Review by Hamitix — APPROVED** [review]

**Pull request merged**